packages/astro/components/ViewTransitions.astro

```diff
@@ -30,6 +30,15 @@ const { fallback = 'animate' } = Astro.props as Props;
 		!!document.querySelector('[name="astro-view-transitions-enabled"]');
 	const triggerEvent = (name: Events) => document.dispatchEvent(new Event(name));
 	const onPageLoad = () => triggerEvent('astro:page-load');
+	const announce = () => {
+		let div = document.createElement('div');
+		div.setAttribute('aria-live', 'assertive');
+		div.setAttribute('aria-atomic', 'true');
+		div.setAttribute('style', 'position:absolute;left:0;top:0;clip:rect(0 0 0 0);clip-path:inset(50%);overflow:hidden;white-space:nowrap;width:1px;height:1px');
+		let title = document.title || document.querySelector('h1')?.textContent || location.pathname;
+		div.textContent = title;
+		document.body.append(div);
```

Comment:
What I believe is happening is since this node is supposedly created for each page on load with its according title, there's not a change to be announced by `aria-live` in the first place. 

I've tried a few changes to fix this, note that doing this **DOES NOT SOLVE IT** (maybe JavaScript is doing some optimization and forcing the text to be added before appended?):
```js
document.body.append(div);
let title = document.title || document.querySelector('h1')?.textContent || location.pathname;
div.textContent = title;
```

What **DID WORK**, was forcing the change to happen only after the node is for sure appended:

```js
document.body.append(div);
setTimeout(() => {
	let title = document.title || document.querySelector('h1')?.textContent || location.pathname;
	div.textContent = title;
}, 50);
```

Reply:
Interesting, thanks will investigate further.

Reply:
@Yan-Thomas updated the implementation with your suggestion but omitted the 50 ms. Curious if that long of a wait is needed or not.

Reply:
Without the wait, or even, smaller waits, the behavior starts to get confusing and sometimes it is announced, some others not. I found a [helpful issue comment with explanation](https://github.com/nvaccess/nvda/issues/8873#issuecomment-432935068), TL;DR screen readers are coalescing the creation and update events and for that reason, it doesn't get announced as a change. Also, I'd even recommend going up to `60` so it's on the same refresh rate as browsers as suggested by the comment.